contracts/registration/TSSSigner.sol

```diff
@@ -11,17 +11,28 @@ abstract contract TSSSigner {
     uint256 public constant P = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F;
 
     address public signer;
+    string public chainName;
 
-    mapping(uint256 => bool) internal _nonces;
+    mapping(uint8 => uint256) internal _nonces;
+    mapping(uint8 => mapping(uint256 => bool)) internal _usedNonces;
 
-    function __TSSSigner_init(address signer_) internal {
+    function __TSSSigner_init(address signer_, string calldata chainName_) internal {
         signer = signer_;
+        chainName = chainName_;
     }
 
-    function _useNonce(uint256 nonce_) internal {
-        require(!_nonces[nonce_], "TSSSigner: nonce used");
+    function getNonce(uint8 methodId_) external view returns (uint256) {
+        return _nonces[methodId_];
+    }
+
+    function _getAndIncrementNonce(uint8 methodId_) internal returns (uint256) {
+        return _nonces[methodId_]++;
+    }
+
+    function _useNonce(uint8 methodId_, uint256 nonce_) internal {
```

Comment:
Since we are now using incremental nonces that are read on-chain, do we need this function?

Reply:
Yes, we need it at least for the `changeICAOMasterTreeRoot` function

Since the timestamp is used there as a nonce